Extension description from user input leads to broken `jlpm build`

## Description

Following the instructions in the [extension tutorial](https://jupyterlab.readthedocs.io/en/stable/extension/extension_tutorial.html), during the `copier copy https://github.com/jupyterlab/extension-template .` step, one is prompted to enter an extension description. If the user enters a single quote anywhere in the description (e.g. as an apostrophe), that will break the extension build when running `pip install -ve .` later in the instructions.
## Reproduce

<!--Describe step-by-step instructions to reproduce the behavior-->

1. Follow [the extension tutorial](https://jupyterlab.readthedocs.io/en/stable/extension/extension_tutorial.html)
2. Run `copier` as instructed.
3. Add an extension description that includes a single quote (`'`) anywhere within.
4. Proceed through the instructions to the build step which has you run `pip install -ve .`
5. Observe the following error during build:

``` 
INFO:hatch_jupyter_builder.utils:> /private/var/folders/yx/49ml7j3s7t31pwd25pyj6rb800n42n/T/pip-build-env-tcyq9c22/overlay/bin/jlpm run install:extension
  src/index.ts(11,59): error TS1005: ',' expected.
```

Inspecting `src/index.ts` at the given line number and character reveals the single quote breaking the `description` k:v pair.

## Expected behavior

Single quotes provided by user in the description (or any user-submitted text) should either be escaped or stripped.

## Context

- Operating System and version: MacOS Ventura
- Browser and version: Chrome 117
- JupyterLab version: 4.0.6
- Extension template version: 4.2.1

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -62,6 +62,50 @@ jobs:
           popd
           rm -rf ${NAME}
 
+  quoted-description:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # This will be used by the base setup action
+        python-version: ["3.11"]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Base Setup
+        uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
+
+      - name: Install dependencies
+        run: |
+          python -m pip install "copier>=8.0.0" jinja2-time "pydantic<2.0.0"
+
+      - name: Setup Git
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+
+      - name: Create pure frontend extension
+        run: |
+          set -eux
+          mkdir myextension
+          python -m copier copy -l -d author_name="My Name" -d description="Let's \"rock and roll\"" -d repository="https://github.com/test/lab-extension" --vcs-ref HEAD --UNSAFE . myextension
+          pushd myextension
+          pip install "jupyterlab>=4.0.0,<5"
+          YARN_ENABLE_IMMUTABLE_INSTALLS=false jlpm
+          jlpm lint:check
+          pip install -e .
+          jupyter labextension develop . --overwrite
+          jupyter labextension list
+          jupyter labextension list 2>&1 | grep -ie "myextension.*OK"
+          python -m jupyterlab.browser_check
+
+          jupyter labextension uninstall myextension
+          pip uninstall -y myextension jupyterlab
+
+          popd
+          rm -rf myextension
+
   no-tests:
     runs-on: ubuntu-latest
     strategy:

--- a/template/package.json.jinja
+++ b/template/package.json.jinja
@@ -1,7 +1,7 @@
 {
     "name": "{{ labextension_name }}",
     "version": "0.1.0",
-    "description": "{{ project_short_description }}",
+    "description": "{{ project_short_description | replace('"', '\\"') }}",
     "keywords": [
         "jupyter",
         "jupyterlab",

--- a/template/src/index.ts.jinja
+++ b/template/src/index.ts.jinja
@@ -14,7 +14,7 @@ import { requestAPI } from './handler';{% endif %}
  */
 const plugin: JupyterFrontEndPlugin<void> = {
   id: '{{ labextension_name }}:plugin',
-  description: '{{ project_short_description }}',
+  description: '{{ project_short_description | replace("'", "\\'") }}',
   autoStart: true,{% if kind.lower() == 'theme' %}
   requires: [IThemeManager],{% endif %}{% if has_settings %}
   optional: [ISettingRegistry],{% endif %}

